File: content/courses/parallel_computing_introduction/codes/random_walk.f90
program random_walk
use random
implicit none

   integer              :: nargs
   character(len=10)    :: ns,nexp
   integer              :: m, n, ntrials, nsteps, step
   real, dimension(2)   :: pos, move
   real                 :: distance, avg

   nargs=command_argument_count()
   if ( nargs .ne. 1 ) then
      write(*,*) 0, 0., 0.
      stop
   else
      call get_command_argument(1,ns)
      read(ns,'(i10)') nsteps
   endif

   call set_random_seed()
   pos=[0.,0.]

   avg=0.

   do n=1,nsteps

      step=randint(1,4)
      if (step==1) then
         move=[0.,1.]
      else if (step==2) then
         move=[0.,-1.]
      else if (step==3) then
         move=[1.,0.]
      else
         move=[-1.,0.]
      endif

      pos=pos+move
   enddo

   distance=sqrt(pos(1)**2+pos(2)**2)

   write(*,*) nsteps, sqrt(real(nsteps)), distance

end program

File: content/courses/parallel-computing-introduction/codes/random_walk.py
import numpy as np
import random
import sys

if len(sys.argv)>1:
   Nsteps=int(sys.argv[1])
else:
   print(0, 0., 0.)
   sys.exit()

pos=np.array([0,0])

moves=[np.array([0,1]),np.array([0,-1]),np.array([1,0]),np.array([-1,0])]

for n in range(Nsteps):
    move=random.choice(moves)
    pos+=move

distance=np.sqrt(pos[0]**2+pos[1]**2)
print("{:.2f},{:.2f},{:.2f}".format(Nsteps, np.sqrt(Nsteps), distance))
